File: app/src/main/java/com/example/testlastfm/ui/albumDetails/AlbumDetailsViewModel.java
package com.example.testlastfm.ui.albumDetails;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.example.testlastfm.model.Album;
import com.example.testlastfm.repository.SearchRepository;

import javax.inject.Inject;

public class AlbumDetailsViewModel extends ViewModel {
    // TODO: Implement the ViewModel

    final MutableLiveData<Album> album;

    SearchRepository searchRepository;

    @Inject
    public AlbumDetailsViewModel(SearchRepository searchRepository) {

        album = new MutableLiveData<>();
        this.searchRepository = searchRepository;
    }

    public LiveData<Album> loadAlbumById(Long albumId) {
        return searchRepository.loadAlbum(albumId);
    }

}

File: app/src/main/java/com/example/testlastfm/ui/albumDetails/AlbumDetailsViewModel.kt
package com.example.testlastfm.ui.albumDetails

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import com.example.testlastfm.model.Album
import com.example.testlastfm.repository.SearchRepository
import javax.inject.Inject

class AlbumDetailsViewModel @Inject constructor(
    private var searchRepository: SearchRepository
) : ViewModel() {
    // TODO: Implement the ViewModel

    internal val album: MutableLiveData<Album> = MutableLiveData()

    fun loadAlbumById(albumId: Long?): LiveData<Album?>? {
        return searchRepository.loadAlbum(albumId)
    }
}